test/lib/sdkdriver/src/testdriver/browser/docker.ts

```diff
@@ -0,0 +1,56 @@
+import { createDocker } from '../../util';
+import BrowserDriver from './base';
+
+/**
+ * Docker browser process driver.
+ * @class
+ * @classdesc A {@link DockerBrowserDriver} manages a Docker browser process
+ *   where the cross-browser test is running.
+ */
+export default class DockerBrowserDriver extends BrowserDriver {
+  private _docker: any;
+
+  /**
+   * Constructor.
+   * @param {"chrome" | "firefox"} browser
+   * @param {string} sdkUrl
+   */
+  constructor(browser: 'chrome' | 'firefox', sdkUrl: string) {
+    super({
+      browser,
+      host: process.env.DOCKER_HOST_IP || 'localhost',
+      params: { sdkUrl }
+    });
+    this._docker = null;
+  }
+
+  /**
+   * Start the browser process.
+   * @param {string} url - URL for running the SDK setup code
+   * @returns {Promise<void>}
+   */
+  startBrowser(url: string): Promise<void> {
```

Comment:
Can this be private?

Reply:
@markandrus TypScript does not allow `abstract` methods to be `private`. So I have to override with public methods.